### PR TITLE
Refactor MonthPaid; Add validation for invalid dates for MonthPaid

### LINKED ISSUE #61: Add test cases and checking for MonthsPaid that test for valid dates
Valid YYYY-MM: XXXX-[01-12]

### LINKED ISSUE #110: Add validation for MonthPaid

### PR DESCRIPTION
**Pull request by sumomomomomo**
Fixes #110
Fixes #61

### CODE REVIEW

**Review by hooyushing — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/model/person/MonthPaid.java [hunk 1]
```diff
@@ -8,16 +11,16 @@
 public class MonthPaid implements Comparable<MonthPaid> {
     public static final String MESSAGE_CONSTRAINTS = "MonthPaid should take a string with the pattern"
             + " YYYY-MM, where one or of such patterns can be included if separated by a space.";
-    public static final String SPLIT_VALIDATION_REGEX = "^[0-9]{4}-[0-9]{2}$";
-    public final String value;
-
-    public MonthPaid(String monthsPaid) {
-        this.value = monthsPaid;
-    }
+    public static final String SPLIT_VALIDATION_REGEX = "^[0-9]{4}-(0[1-9]|1[0-2])$";
```
Comment (hooyushing) [inline src/main/java/seedu/address/model/person/MonthPaid.java:14]: We are allowing 0000 as a valid year?

File: src/main/java/seedu/address/model/person/MonthPaid.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
  */
 public class MonthPaid implements Comparable<MonthPaid> {
     public static final String MESSAGE_CONSTRAINTS = "MonthPaid should take a string with the pattern"
-            + " YYYY-MM, where one or of such patterns can be included if separated by a space.";
+            + " YYYY-MM, where MM should be between 01 and 12.";
     public static final String SPLIT_VALIDATION_REGEX = "^[0-9]{4}-(0[1-9]|1[0-2])$";
```
Comment (hooyushing) [inline src/main/java/seedu/address/model/person/MonthPaid.java:14]: Should restrict years to 1900 - 2200 AC

Comment (sumomomomomo) [reply src/main/java/seedu/address/model/person/MonthPaid.java]: Fixed

**Review by hooyushing — COMMENTED** [review]
LGTM

File: src/main/java/seedu/address/model/person/MonthPaid.java (hunk 1 above)
Comment (sumomomomomo) [reply src/main/java/seedu/address/model/person/MonthPaid.java]: Fixed

**Pull request merged**